### DISCUSSION TITLE
How about general rotations?

### DISCUSSION THREAD

**Discussion by AFLuo** [tuneinsight/lattigo#430] [Q&A]
Existing rotation methods for `heint` includes `RotateColumnsNew` and `RotateRowsNew` to rotate columns and rows respectively, which is quite strange since such rotation methods put restrictions on dimensions. I wonder if i can rotate as if there is only one row.
Take an example, I have a vector, which contains `params.N()` elements, and i want to rotate over the ct encrypting this vector.
It seems that existing rotation methods does not support this operation.

**Comment by Pro7ech** [comment]
@AFLuo, the plaintext domain for batched vectors in BFV and BGV is `Z_t^{2 * N/2}`, i.e. a matrix of dimension `2 x N/2` with coefficients modulo `t`. Rotations are defined on the rows and columns of this matrix. So what you are asking for is not possible because the plaintext algebra does not allow it.

**Reply by AFLuo** [reply]
Got it,  thank you very much
